src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,233 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
-    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
-    if (!currentItemPosition || !navigationMenuPosition) {
-      return;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPath]);
+
+  /**
+   * Ensures that the cursor is synched with the currently selected item.
+   * Scrolls selected item into view.
+   */
+  useEffect(() => {
+    if (currentNodeId.current) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === currentNodeId.current);
     }
-    // If the current item is not visible, scroll the item into view.
-    if (currentItemPosition.bottom > navigationMenuPosition.bottom || currentItemPosition.top < navigationMenuPosition.top) {
-      selectedItem.current.scrollIntoView();
+    if (previousSelectedPath.current !== selectedPath) {
+      const selectedItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+      const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
+      if (selectedItemPosition && navigationMenuPosition && (selectedItemPosition.bottom > navigationMenuPosition.bottom || selectedItemPosition.top < navigationMenuPosition.top)) {
+        selectedItem.current.scrollIntoView();
+      }
+      previousSelectedPath.current = selectedPath;
     }
-  });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [openKeys]);
+
+  /**
+   * Ensures the parent of the currently selected item is expanded.
+   */
+  useEffect(() => {
+    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPath]);
 
   const handleOnClick = (event, item) => {
+    // Don't update state if function is called through handleKeyDown
+    if (event.type === 'click') {
+      const eventTargetId = event.target.getAttribute('id');
+      currentNodeId.current = eventTargetId;
+      cursor.current = visibleNodes.findIndex((el) => el.id === eventTargetId);
+      focusCurrentNode();
+    }
     if (!item.childItems) {
       onSelect(item.path);
       return;
     }
     setOpenKeys({ ...openKeys, [item.path]: !openKeys[item.path] });
   };
 
+  const handleDownArrow = () => {
+    if (cursor.current + 1 < visibleNodes.length) {
+      cursor.current += 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  const handleUpArrow = () => {
+    if (cursor.current >= 1) {
+      cursor.current -= 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds parent of the current node, used for left arrow functionality
+   */
+  const findParentNode = () => {
+    if (!currentNodeId.current) {
+      return;
+    }
+    const parentId = visibleNodes.find((el) => el.id === currentNodeId.current).parent;
+    if (parentId !== '') {
```

Comment:
I'd recommend setting parent id to `undefined` or an explicit `null` value when at the root of the node tree.

Reply:
Fixed this in a recent commit